src/CoreWCF.Primitives/src/CoreWCF/Security/ReceiveSecurityHeader.cs

```diff
@@ -0,0 +1,2187 @@
+                    case ReceiveSecurityHeaderElementCategory.EncryptedKey:
+                    case ReceiveSecurityHeaderElementCategory.EncryptedData:
+                    case ReceiveSecurityHeaderElementCategory.SignatureConfirmation:
+                    case ReceiveSecurityHeaderElementCategory.SecurityTokenReference:
+                        // no op
+                        break;
+                    default:
+                        Fx.Assert("invalid element category");
+                        break;
+                }
+            }
+        }
+
+        internal void ExecuteSubheaderDecryptionPass()
+        {
+            for (int position = 0; position < this.elementManager.Count; position++)
+            {
+                if (this.elementManager.GetElementCategory(position) == ReceiveSecurityHeaderElementCategory.EncryptedData)
+                {
+                    EncryptedData encryptedData = this.elementManager.GetElement<EncryptedData>(position);
+                    bool dummy = false;
+                    ProcessEncryptedData(encryptedData, this.timeoutHelper.RemainingTime(), position, false, ref dummy);
+                }
+            }
+        }
+
+        internal void ExecuteReadingPass(XmlDictionaryReader reader)
+            {
+            int position = 0;
+            while (reader.IsStartElement())
+            {
+                if (IsReaderAtSignature(reader))
+                {
+                    ReadSignature(reader, AppendPosition, null);
+                }
+                else if (IsReaderAtReferenceList(reader))
+                {
+                    ReadReferenceList(reader);
+                }
+                else if (this.StandardsManager.WSUtilitySpecificationVersion.IsReaderAtTimestamp(reader))
+                {
+                    ReadTimestamp(reader);
+                }
+                else if (IsReaderAtEncryptedKey(reader))
+                {
+                    ReadEncryptedKey(reader, false);
+                }
+                else if (IsReaderAtEncryptedData(reader))
+                {
+                    ReadEncryptedData(reader);
+                }
+                else if (this.StandardsManager.SecurityVersion.IsReaderAtSignatureConfirmation(reader))
+                {
+                    ReadSignatureConfirmation(reader, AppendPosition, null);
+                }
+                else if (IsReaderAtSecurityTokenReference(reader))
+                {
+                    ReadSecurityTokenReference(reader);
+                }
+                else
+                {
+                    ReadToken(reader, AppendPosition, null, null, null, this.timeoutHelper.RemainingTime());
+                }
+                position++;
+            }
+
+            reader.ReadEndElement(); // wsse:Security
+            reader.Close();
+        }
+
+        internal void ExecuteFullPass(XmlDictionaryReader reader)
+        {
+            try
```

Comment:
Remove try/catch block. I presume this was for debugging

Reply:
you are right..